pkg/apps/generated/listers/apps/internalversion/deploymentconfig_expansion.go

```diff
@@ -0,0 +1,82 @@
+package internalversion
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/v1"
+
+	apps "github.com/openshift/origin/pkg/apps/apis/apps"
+	imageapi "github.com/openshift/origin/pkg/image/apis/image"
+)
+
+// DeploymentConfigListerExpansion allows custom methods to be added to
+// DeploymentConfigLister.
+type DeploymentConfigListerExpansion interface {
+	GetConfigForController(rc *v1.ReplicationController) (*apps.DeploymentConfig, error)
+	GetConfigsForImageStream(stream *imageapi.ImageStream) ([]*apps.DeploymentConfig, error)
+}
+
+// DeploymentConfigNamespaceListerExpansion allows custom methods to be added to
+// DeploymentConfigNamespaceLister.
+type DeploymentConfigNamespaceListerExpansion interface{}
+
+// GetConfigForController returns the managing deployment config for the provided replication controller.
+func (s *deploymentConfigLister) GetConfigForController(rc *v1.ReplicationController) (*apps.DeploymentConfig, error) {
+	dcName := rc.Annotations[apps.DeploymentConfigAnnotation]
+	obj, exists, err := s.indexer.GetByKey(rc.Namespace + "/" + dcName)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, errors.NewNotFound(apps.Resource("deploymentconfig"), dcName)
+	}
+	return obj.(*apps.DeploymentConfig), nil
+}
+
+// GetConfigsForImageStream returns all the deployment configs that point to the provided image stream
+// by searching through using the ImageStreamReferenceIndex (deployment configs are indexed in the cache
+// by namespace and by image stream references).
+func (s *deploymentConfigLister) GetConfigsForImageStream(stream *imageapi.ImageStream) ([]*apps.DeploymentConfig, error) {
+	items, err := s.indexer.ByIndex(ImageStreamReferenceIndex, stream.Namespace+"/"+stream.Name)
```

Comment:
This index was never added before, so all calls to this method in the legacy package returned an error!